File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.c
int largestAltitude(int* gain, int gainSize){
    int *a = malloc(sizeof(int) * (gainSize+1));
    int i = 0,max = 0,sum = 0;
    
    a[0] = 0;
    while(i < gainSize)
    {
        sum += gain[i];
        a[i+1] = sum;
        if(sum > max)
            max = sum;
        i++;
    }
    return max;
}

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int max = 0;
        int tmp = 0;
        
        for(int i = 0; i < gain.length; i++){
            tmp += gain[i];
            if(tmp > max)
                max = tmp;
        }
        return max;
    }
}